Dev/scripting and remote control

cameroncros opened the pull request.

cameroncros committed:
Added scripting (not quite feature complete) and added Telnet server to the app so the drone can have commands issued remotely

cameroncros committed:
Create scripting.md

Added basic documentation for the scripting system

cameroncros committed:
Update scripting.md

Fixed formatting

cameroncros committed:
Added missing commands

cameroncros committed:
Merge remote-tracking branch 'origin/dev/ScriptingAndRemoteControl' into dev/ScriptingAndRemoteControl

cameroncros committed:
Added additional commands, and updated the documentation

cameroncros committed:
Updated documentation

cameroncros committed:
Added HELP and GETHOME commands

cameroncros committed:
Updated documentation for scripting interface

cameroncros committed:
Removed roll and pitch variables from commands. Not used anyway

cameroncros committed:
fixed the exception handler so it only catches numberformatexception

7178301 committed:
FIX: Telnet Command Function Name Continuity

cameroncros committed:
Moved TelnetServer code to eagleAPIAndroid

cameroncros committed:
Merge remote-tracking branch 'origin/dev/ScriptingAndRemoteControl' into dev/ScriptingAndRemoteControl

cameroncros committed:
Added LOG command to api

cameroncros committed:
Added tests for relative commands, removed unnecessary includes

cameroncros committed:
Renamed Absolute commands to GPS commands to be more clear.

cameroncros committed:
Fixed simulator class so it works correctly.
Fixed Position and Bearing classes and tests

cameroncros committed:
Deleted Flight Tests and saved Routes from sparrow

cameroncros committed:
Refactored TelnetServer code

cameroncros committed:
Made the telnet server say when the command is invalid instead of repeating the command

cameroncros committed:
Moved toString to toPrettyString and reimplemented toString in a more basic format

7178301 commented:
SDKAdaptor.java - goHome(double speed) 
incorrect default systax. please check goHome for correct syntax.

Drone.java - executeInstruction() runRoutine() join() InvalidInstructionException
Could these please be moved in a separate class called messaging/command/control.    Drone should be for fundamental core  functionality and not look messy. It will be the first class a user looks at so needs to be neat.

Position.java Should there be different position classes. One for GPS and one for relative or is there going to be some conversion/check between meters and GPS?

cameroncros committed:
fixed getHome

cameroncros committed:
Extracted scripting code to seperate class, ScriptingEngine

cameroncros commented:
The original getHome was incorrect as well, but i have fixed it.

I have also extracted the script sections to a ScriptingEngine class.

The different position types really needs a lot more discussion. I created a branch to see if it is feasable, and it looks doable with a minor amount of modifications

cameroncros committed:
Fixed threading for TelnetServer, and removed need for AndroidTelnetServer

cameroncros committed:
Fixed gohome, and simulator class. Moved scripting engine tests to correct class

7178301 committed:
FIX: Position Split Into PositionMetric & PositionGPS

7178301 committed:
QUALITY: Update TelnetServer Javadoc

7178301 committed:
Merge branch 'development' into dev/ScriptingAndRemoteControl

Conflicts:
	eagle/eagleAPI/src/main/java/eagle/Drone.java
	eagle/eagleAPI/src/main/java/eagle/Log.java
	eagle/eagleAPI/src/main/java/eagle/navigation/positioning/Angle.java
	eagle/eagleAPI/src/main/java/eagle/navigation/positioning/Position.java
	eagle/eagleAPI/src/main/java/eagle/sdkInterface/SDKAdaptor.java
	eagle/eagleAPI/src/main/java/eagle/sdkInterface/sdkAdaptors/Simulator/Simulator.java
	eagle/eagleAPIAndroid/src/main/java/eagle/sdkInterface/LogAndroid.java
	eagle/eagleAPIAndroid/src/main/java/eagle/sdkInterface/sdkAdaptors/DJI/Phantom2Vision.java
	eagle/eagleAPIAndroid/src/main/java/eagle/sdkInterface/sdkAdaptors/Flyver/F450Flamewheel.java
	eagle/sparrow/src/main/java/au/edu/swin/sparrow/APIAdaptorActivity.java
	eagle/sparrow/src/main/java/au/edu/swin/sparrow/DroneAdaptor/FlightTest.java
	eagle/sparrow/src/main/java/au/edu/swin/sparrow/DroneAdaptor/FlightTests/FlyUpFlyDown.java
